diff --git a/node_modules/dd-trace/packages/datadog-plugin-graphql/src/index.js b/node_modules/dd-trace/packages/datadog-plugin-graphql/src/index.js
index 99820a2..c8ffee5 100644
--- a/node_modules/dd-trace/packages/datadog-plugin-graphql/src/index.js
+++ b/node_modules/dd-trace/packages/datadog-plugin-graphql/src/index.js
@@ -48,6 +48,7 @@ function createWrapParse (tracer, config) {
 
       analyticsSampler.sample(span, config.analytics)
 
+      let err = null;
       try {
         const document = parse.apply(this, arguments)
         const operation = getOperation(document)
@@ -60,12 +61,12 @@ function createWrapParse (tracer, config) {
 
         addDocumentTags(span, document)
 
-        finish(null, span)
-
         return document
       } catch (e) {
-        finish(e, span)
+        err = e
         throw e
+      } finally {
+        finish(err, span)
       }
     }
   }
