patch-package
--- a/node_modules/dd-trace/src/format.js
+++ b/node_modules/dd-trace/src/format.js
@@ -57,10 +57,18 @@ function extractTags (trace, span) {
       case 'error.msg':
       case 'error.stack':
         trace.error = 1
-        trace.meta[tag] = String(tags[tag])
+        try {
+          trace.meta[tag] = String(tags[tag])
+        } catch (e) {
+          console.error("error:", e.message, "tag:", tag, "value:", tags[tag], "prototype:", tags[tag] ? tags[tag].__proto__ : undefined)
+        }
         break
       default:
-        trace.meta[tag] = String(tags[tag])
+        try {
+          trace.meta[tag] = String(tags[tag])
+        } catch (e) {
+          console.error("error:", e.message, "tag:", tag, "value:", tags[tag], "prototype:", tags[tag] ? tags[tag].__proto__ : undefined)
+        }
     }
   })
 }
