patch-package
--- a/node_modules/dd-trace/src/plugins/http.js
+++ b/node_modules/dd-trace/src/plugins/http.js
@@ -3,6 +3,7 @@
 const url = require('url')
 const opentracing = require('opentracing')
 const semver = require('semver')
+const platform = require("../platform")
 
 const Tags = opentracing.Tags
 const FORMAT_HTTP_HEADERS = opentracing.FORMAT_HTTP_HEADERS
@@ -43,7 +44,17 @@ function patch (http, methodName, tracer, config) {
 
       const req = request.call(this, options, callback)
 
-      req.on('socket', () => {
+      const startedConnectionAt = platform.now()
+      req.on("socket", socket => {
+        // TODO: We only make https connections, but if we ever were to make
+        //       plain http requests then we should trace the `connect` event.
+        socket.on("secureConnect", () => {
+          tracer.startSpan("http.request.connect", {
+            childOf: span,
+            startTime: startedConnectionAt
+          }).finish()
+        })
+
         // empty the data stream when no other listener exists to consume it
         if (req.listenerCount('response') === 1) {
           req.on('response', res => res.resume())
