patch-package
--- a/node_modules/@types/graphql-relay/index.d.ts
+++ b/node_modules/@types/graphql-relay/index.d.ts
@@ -72,6 +72,7 @@ export interface ConnectionConfig {
     resolveCursor?: GraphQLFieldResolver<any, any> | null;
     edgeFields?: Thunk<GraphQLFieldConfigMap<any, any>> | null;
     connectionFields?: Thunk<GraphQLFieldConfigMap<any, any>> | null;
+    interfaces?: [GraphQLInterfaceType];
 }
 
 export interface GraphQLConnectionDefinitions {
@@ -211,11 +212,11 @@ export function getOffsetWithDefault(
 
 // mutation/mutation.js
 
-export type mutationFn = (
-    object: any,
-    ctx: any,
+export type mutationFn<TInput, TOutput, TContext> = (
+    object: TInput,
+    ctx: TContext,
     info: GraphQLResolveInfo
-) => Promise<any> | any;
+) => Promise<TOutput> | TOutput;
 
 /**
  * A description of a mutation consumable by mutationWithClientMutationId
@@ -231,12 +232,12 @@ export type mutationFn = (
  * input field, and it should return an Object with a key for each
  * output field. It may return synchronously, or return a Promise.
  */
-export interface MutationConfig {
+export interface MutationConfig<TInput, TOutput, TContext> {
     name: string;
     description?: string;
     inputFields: Thunk<GraphQLInputFieldConfigMap>;
-    outputFields: Thunk<GraphQLFieldConfigMap<any, any>>;
-    mutateAndGetPayload: mutationFn;
+    outputFields: Thunk<GraphQLFieldConfigMap<TOutput, TContext>>;
+    mutateAndGetPayload: mutationFn<TInput, TOutput, TContext>;
     deprecationReason?: string;
 }
 
@@ -244,8 +245,8 @@ export interface MutationConfig {
  * Returns a GraphQLFieldConfig for the mutation described by the
  * provided MutationConfig.
  */
-export function mutationWithClientMutationId(
-    config: MutationConfig
+export function mutationWithClientMutationId<TInput, TOutput, TContext>(
+    config: MutationConfig<TInput, TOutput, TContext>
 ): GraphQLFieldConfig<any, any>;
 
 // node/node.js
