patch-package
--- a/node_modules/@types/graphql/type/definition.d.ts
+++ b/node_modules/@types/graphql/type/definition.d.ts
@@ -319,15 +319,15 @@ export interface GraphQLScalarTypeConfig<TInternal, TExternal> {
  *     });
  *
  */
-export class GraphQLObjectType {
+export class GraphQLObjectType<TSource, TContext> {
     name: string;
     description: Maybe<string>;
     astNode: Maybe<ObjectTypeDefinitionNode>;
     extensionASTNodes: Maybe<ReadonlyArray<ObjectTypeExtensionNode>>;
-    isTypeOf: Maybe<GraphQLIsTypeOfFn<any, any>>;
+    isTypeOf: Maybe<GraphQLIsTypeOfFn<TSource, TContext>>;
 
-    constructor(config: GraphQLObjectTypeConfig<any, any>);
-    getFields(): GraphQLFieldMap<any, any>;
+    constructor(config: GraphQLObjectTypeConfig<TSource, TContext>);
+    getFields(): GraphQLFieldMap<TSource, TContext>;
     getInterfaces(): GraphQLInterfaceType[];
     toString(): string;
     toJSON(): string;
@@ -371,7 +371,9 @@ export interface GraphQLResolveInfo {
     readonly path: ResponsePath;
     readonly schema: GraphQLSchema;
     readonly fragments: { [key: string]: FragmentDefinitionNode };
-    readonly rootValue: any;
+    // This is re-typed as `void` to ensure nobody at Artsy uses the old pattern
+    // anymore before we moved to using the proper `context` parameter.
+    readonly rootValue: void;
     readonly operation: OperationDefinitionNode;
     readonly variableValues: { [variableName: string]: any };
 }
@@ -499,13 +501,14 @@ export interface GraphQLInterfaceTypeConfig<TSource, TContext> {
  *     });
  *
  */
-export class GraphQLUnionType {
+// TODO: Remove `any` defaults before submitting upstream.
+export class GraphQLUnionType<TSource = any, TContext = any> {
     name: string;
     description: Maybe<string>;
     astNode?: Maybe<UnionTypeDefinitionNode>;
-    resolveType: Maybe<GraphQLTypeResolver<any, any>>;
+    resolveType: Maybe<GraphQLTypeResolver<TSource, TContext>>;
 
-    constructor(config: GraphQLUnionTypeConfig<any, any>);
+    constructor(config: GraphQLUnionTypeConfig<TSource, TContext>);
 
     getTypes(): GraphQLObjectType[];
 
